torchtext/csrc/gpt2_bpe_tokenizer.cpp

```diff
@@ -0,0 +1,51 @@
+#include <algorithm>
+#include <gpt2_bpe_tokenizer.h>
+#include <regex.h> // @manual
+
+namespace torchtext {
+    const Regex kGPT2Regex(
+        "(\\'s|\\'t|\\'re|\\'ve|\\'m|\\'ll|\\'d| ?\\pL+|"
+        " ?\\pN+| ?[^\\s\\v\\pL\\pN]+|[\\s\\v]+)"
+    );
+
+    bool is_whitespace(const std::string &input) {
+        for (std::string::const_iterator it = input.begin(); it != input.end(); ++it) {
```

Comment:
nit: Range-based for loop might improve the readability.

https://stackoverflow.com/a/9438329
https://en.cppreference.com/w/cpp/language/range-for

Reply:
Done